src/main/java/org/openmrs/addonindex/domain/AddOnInfoAndVersions.java

```diff
@@ -192,4 +192,9 @@ public void addTag(String tag) {
 		}
 		tags.add(tag);
 	}
+
+	public void setDetailsBasedOnLatestVersion(AddOnVersion version){
```

Comment:
Move the logic for picking the latest version of this module into this method. I.e. it should take no arguments, and it should get the latest version itself.